UserApi/UserApi/Helper/UserProfileHelper.cs

```diff
@@ -58,17 +65,17 @@ public async Task<UserProfile> GetUserProfile(string filter)
             return response;
         }
 
-        private List<string> GetUserCaseTypes(List<AdGroup> userGroups)
-        {
-            return userGroups.Where(IsCaseType).Select(g => g.ToString()).ToList();
-        }
+        private static bool IsCaseType(AdGroup group) => CaseTypeMappings.ContainsKey(group);
 
-        private bool IsCaseType(AdGroup group)
+        private static List<string> GetUserCaseTypes(IEnumerable<AdGroup> userGroups)
         {
-            return group == AdGroup.FinancialRemedy || group == AdGroup.MoneyClaims;
+            return userGroups
+                .Where(IsCaseType)
+                .Select(c => CaseTypeMappings[c])
+                .ToList();
         }
 
-        private UserRole GetUserRole(List<AdGroup> userGroups)
+        private static UserRole GetUserRole(List<AdGroup> userGroups)
```

Comment:
Pass param as IEnumberable because the methods acts on IEnumerable e.g. Contains(), Any()

Reply:
Shouldn't enumerate an enumerable several times (as it would re-run the code behind it time and time again) that's why I use a collection